packages/adapters-library/src/adapters/solv/products/yield-market/solvYieldMarketAdapter.ts

```diff
@@ -266,159 +277,146 @@ export class SolvYieldMarketAdapter implements IProtocolAdapter {
     return position
   }
 
-  private async getRedemptions(
-    userAddress: string,
-  ): Promise<ProtocolPosition[]> {
-    const { redemptionDelegateAddress: openFundRedemptionDelegateAddress } =
-      this.yieldMarketConfig
-
-    // Count every instance of the GOEFS that the user holds
-    const tokensCount =
-      await this.openFundRedemptionDelegateContract['balanceOf(address)'](
-        userAddress,
-      )
-    if (!tokensCount) return []
-
-    const indexes = [...Array(Number.parseInt(tokensCount.toString())).keys()]
-
-    // Each GOEFR the user holds represents a position
-    return filterMapAsync(indexes, async (index) =>
-      this.getRedemption(userAddress, openFundRedemptionDelegateAddress, index),
-    )
-  }
-
   /**
-   * Build the position represented by the Nth GOEFR that the user holds
+   * Returns a promise that resolves to the list of tokenIds (bigints) that
+   * the user holds for either the GOEFS or the GOEFR.
+   *
+   * Use the flag `sftType` to choose between GOEFS and GOEFR.
    */
-  private async getRedemption(
+  private async getTokenIds(
     userAddress: string,
-    sftAddress: string,
-    index: number,
-  ): Promise<ProtocolPosition | undefined> {
-    const tokenId =
-      await this.openFundRedemptionDelegateContract.tokenOfOwnerByIndex(
-        userAddress,
-        index,
-      )
-    const balance =
-      await this.openFundRedemptionDelegateContract['balanceOf(uint256)'](
-        tokenId,
-      )
+    sftType: 'share' | 'redemption' = 'share',
+  ): Promise<bigint[]> {
+    const contract =
+      sftType === 'share'
+        ? this.openFundShareDelegateContract
+        : this.openFundRedemptionDelegateContract
 
-    if (!balance) return
-
-    const decimals =
-      await this.openFundRedemptionDelegateContract.valueDecimals()
-    const slot = await this.openFundRedemptionDelegateContract.slotOf(tokenId)
-    const [poolId, currency] =
-      await this.openFundRedemptionConcreteContract.getRedeemInfo(slot)
-    const [latestSetNavTime] = await this.navOracleContract.poolNavInfos(poolId)
-    const [nav] = await this.navOracleContract.getSubscribeNav(
-      poolId,
-      latestSetNavTime,
-    )
+    const tokensCount = await contract['balanceOf(address)'](userAddress)
 
-    const shareSlot = this.slotReverseLookup(poolId)
-    const name = `${this.getPoolName(
-      shareSlot.toString(),
-    )} | Redemption pending`
+    if (!tokensCount) return []
 
-    const {
-      symbol: underlyingSymbol,
-      decimals: underlyingDecimals,
-      name: underlyingName,
-    } = await this.helpers.getTokenMetadata(currency)
+    const indexes = [...Array(Number.parseInt(tokensCount.toString())).keys()]
```

Comment:
```suggestion
    const indexes = [...Array(Number(tokensCount)).keys()]
```
A bit less verbose and equally unsafe way of casting from BigInt to Number.